docs/getting-started/auth-self-defence-checks.adoc

```diff
@@ -0,0 +1,57 @@
+= Self Defence Checks
+
+As part of the Auth SDK module, there is a self defence check feature.  This feature offers multiple checks that can be used to analyse the secuirty of the device.
+
+== Prequisite's
+
+. Import and initialise the Auth SDK module - link:auth.adoc[AeroGear Services Auth SDK].
+
+== Usage
+
+To begin using the self defence checks in your app, you'll need to retrieve an instance of `SecurityService`. The `SecurityService` can be retrieved using the `MobileCore#getInstance` method.
+
+[source, java]
+----
+SecurityService securityService = MobileCore.getInstance(SecurityService.class);
+----
+
+Any subsequent `MobileCore#getInstance` method will return the same instance of `SecurityService`.
+
+=== Invoking a Single Self Defence Check
+The `SecurityService#check` method can be used to run a single secuirty check.  It expects a `SecurityCheckType` paramater and returns `SecurityCheckResult`.
+
+[source, java]
+----
+SecurityCheckResult result = securityService.check(SecurityCheckType.<check_type>);
+----
+
+See all available `SecurityCheckType` link:auth-self-defence-javadoc.adoc[here].
+
+=== Invoking Multiple Self Defence Checks
+To run multiple self defence checks you'll first need to invoke the `SecurityService#getCheckExecutor` method.  This methods returns `SecurityCheckExecutor` where checks can be added (chained) to the executor.
+
+[source, java]
+----
+SecurityCheckExecutor checkExecutor = securityService.getCheckExecutor();
+----
+
+The `SecurityCheckExecutor#addCheck` method can be used to add multiple self defence checks to the `checkExecutor`.  The `addCheck` function accepts `SecurityCheckType` as a parameter and returns the `SecurityCheckExecutor`. 
+
+[source, java]
+----
+// add two checks to the checkExecutor
+checkExecutor.addCheck(SecurityCheckType.<check_type>).addCheck(SecurityCheckType.<check_type>);
+----
+
+To run all the security self defence checks that have been added to the `checkExecutor` the `SecurityCheckExecutor#execute` method must be invoked.  This method returns an array of `SecurityCheckResult`.
+
+[source, java]
+----
+SecurityCheckResult results[] = checkExecutor.execute();
```

Comment:
This has changed as well. Now depending on the used executor, the outcome is:

- Map<String, SecurityCheckResult> for the sync executor
- Map<String, Future<SecurityCheckResult>> for the async executor

in both cases, the key is the name of the security check that produced the result.